tensorflow_datasets/scripts/metadata_cleanup.py

```diff
@@ -0,0 +1,92 @@
+# Lint as: python3
+"""Script to Cleanup Metadata directory.
+
+To test:
+Goto tensorflow_datasets/scripts and run using python metadata_cleanup.py
+
+```
+cd tensorflow_datasets/scripts
+python metadata_cleanup.py
+```
+It returns :
+
+Total number of registered datasets
+Total number of subdirectories in metadata direcotry
+Total number of meta directories removed 
+"""
+
+from absl import app
+import os
+import re
+import shutil
+import tensorflow as tf
+from tensorflow_datasets.core import registered
+
+metadata_path = os.path.join(os.pardir, "testing/metadata")
+
+def _extract_full_names_from_dirs(metadata_dir):
+    '''Get all metadata direcotry versions paths '''
+    meta_paths = set()
+    for root, dirs, files in os.walk(metadata_dir):
+        for fileName in files:
+            meta_paths.add(os.path.join( root[len(metadata_dir):]))
+    meta_paths.remove('')
+    return list(meta_paths)
+
+def get_registered_version():

```

Comment:
What are you trying to do with this function ? You should not need this and use `iter_dataset_full_names `

Reply:
This function create list of all register versions paths which is exactly matches with format of paths extracted from metadata. I will remove this function but then I have to define it in `delete_metadata_dirs` function